File: tests/csharp/csharp_classes/class_static_method.cs
// vybe-test: csharp/csharp_classes/class_static_method
// origin: languages/csharp/tests/csharp/test_csharp_classes.rs

using static __Harness;

__P((MathUtils.Square(7)).ToString());
__Check("49");

class MathUtils {
    public static int Square(int x) { return x * x; }
}

public static class __Harness {
    public static string __buf = "";
    public static void __P(string s) { __buf = __buf + s + "\n"; }
    public static void __Pr(string s) { __buf = __buf + s; }
    public static void __Check(string want) {
        if (__buf != want && __buf != want + "\n") {
            Console.WriteLine("FAIL: want [" + want + "] got [" + __buf + "]");
            throw new Exception("assertion failed");
        }
    }
}

File: tests/js/ecma_classes/class_static_method.js
// vybe-test: js/ecma_classes/class_static_method
// origin: languages/js/tests/js/test_ecma_classes.rs

function __line(...args) {
    // console.log joins its arguments with a single space. String() is the
    // coercion Vybe's logging host applies to each one.
    return args.map(String).join(" ");
}

function __check(got, want) {
    if (got !== want) {
        console.log("FAIL: want [" + want + "] got [" + got + "]");
        throw new Error("assertion failed");
    }
}

class MathUtils {
    static square(x) { return x * x; }
    static cube(x) { return x * x * x; }
}
__check(__line(MathUtils.square(4)), "16");
__check(__line(MathUtils.cube(3)), "27");
